porcupine/utils.py

```diff
@@ -645,14 +645,18 @@ def callback(shifted: bool, event: tkinter.Event[tkinter.Misc]) -> Any:
 
 # TODO: document this?
 def ask_encoding(text: str, old_encoding: str) -> str | None:
-    label_width = 400
+    if porcupine.get_main_window().tk.call("winfo", "exists", ".choose_encoding"):
+        porcupine.get_main_window().tk.call("focus", ".choose_encoding")
```

Comment:
The existing encoding dialog could be completely unrelated, e.g. you could click the encoding choosing buttons of two different tabs without closing the encoding dialog in between.

```suggestion
        porcupine.get_main_window().tk.call("destroy", ".choose_encoding")
```

Reply:
Wow, you always know some corner case :D
I would have never thought of this one.

Reply:
Or perhaps not so much a corner-case.